src/app/modules/websocket-debug-panel/components/message-list/message-list.component.spec.ts

```diff
@@ -280,5 +280,81 @@ describe('MessageListComponent', () => {
     });
   });
 
+  describe('filter functionality', () => {
```

Comment:
Much of the codebase takes a different approach to unit tests. User DOM interactions are mocked and behavior is verified via DOM queries.

Reply:
Which do you think is best?

Reply:
I think it makes sense for tests to verify what the user "sees" with DOM queries and harnesses.